core/arch/arm/mm/core_mmu.c

```diff
@@ -876,30 +876,48 @@ enum teecore_memtypes core_mmu_get_type_by_pa(paddr_t pa)
 	return map->type;
 }
 
-int core_tlb_maintenance(int op, unsigned int a)
+int __deprecated core_tlb_maintenance(int op, unsigned long a)
 {
 	switch (op) {
 	case TLBINV_UNIFIEDTLB:
-		secure_mmu_unifiedtlbinvall();
+		tlbi_all();
 		break;
 	case TLBINV_CURRENT_ASID:
-		secure_mmu_unifiedtlbinv_curasid();
+#ifdef ARM32
+		tlbi_asid(read_contextidr());
+#endif
+#ifdef ARM64
+		tlbi_asid(read_contextidr_el1());
+#endif
 		break;
 	case TLBINV_BY_ASID:
-		secure_mmu_unifiedtlbinv_byasid(a);
+		tlbi_asid(a);
 		break;
 	case TLBINV_BY_MVA:
-		EMSG("TLB_INV_SECURE_MVA is not yet supported!");
-		while (1)
-			;
-		secure_mmu_unifiedtlbinvbymva(a);
+		panic();
 		break;
 	default:
 		return 1;
 	}
 	return 0;
 }
 
+void tlbi_mva_range(vaddr_t va, size_t size, size_t granule)
+{
+	assert(granule == CORE_MMU_PGDIR_SIZE || granule == SMALL_PAGE_SIZE);
+
+	dsb_ishst();
+	while (size) {
+		tlbi_mva_allasid_nosync(va);
+		if (size < granule)
+			break;
+		size -= granule;
```

Comment:
No it's just that for a longer and more complex function it can become tricky to keep track of what was the input and what has been changed.